CSharpSourceCode/Battle/Dismemberment/Dismemberment.cs

```diff
@@ -0,0 +1,158 @@
+            head.AddPhysics(1f, head.CenterOfMass, head.GetBodyShape(), velocityVec, angularVec, PhysicsMaterial.GetFromName("flesh"), false, -1);
+        }
+        private static void CreateBloodBurst(Agent victim)
+        {
+            MatrixFrame boneEntitialFrameWithIndex = victim.AgentVisuals.GetSkeleton().GetBoneEntitialFrameWithIndex((byte)victim.BoneMappingArray[HumanBone.Head]);
+            Vec3 vec = victim.AgentVisuals.GetGlobalFrame().TransformToParent(boneEntitialFrameWithIndex.origin);
+            victim.CreateBloodBurstAtLimb(13, ref vec, 0.5f + MBRandom.RandomFloat * 0.5f);
+        }
+
+        public static Agent SpawnAgent(Agent agent)
+        {
+            AgentBuildData agentBuildData = new AgentBuildData(agent.Character);
+            agentBuildData.NoHorses(true);
+            agentBuildData.NoWeapons(true);
+            agentBuildData.NoArmor(false);
+            agentBuildData.Team(Mission.Current.PlayerEnemyTeam);
+            agentBuildData.TroopOrigin(agent.Origin);
+
+            MatrixFrame frame = default(MatrixFrame);
+            frame.origin = agent.Position;
+            frame.rotation = agent.Frame.rotation;
+            agentBuildData.InitialFrame(frame);
+
+            return Mission.Current.SpawnAgent(agentBuildData, false, 0);
+        }
+        private static void KillAgent(Agent agent)
+        {
+            if (!GameNetwork.IsClientOrReplay)
+            {
+                Blow blow = new Blow(agent.Index);
+                blow.DamageType = 0;
+                blow.StrikeType = 0;
+                blow.BoneIndex = agent.Monster.HeadLookDirectionBoneIndex;
+                blow.Position = agent.Position;
+                blow.Position.z = blow.Position.z + agent.GetEyeGlobalHeight();
+                blow.BaseMagnitude = 2000f;
+                blow.InflictedDamage = 2000;
+                Vec3 vec = new Vec3(1f, 0f, 0f, -1f);
+
+                if (Mission.Current.InputManager.IsGameKeyDown(2))
+                    vec = new Vec3(-1f, 0f, 0f, -1f);
+                else
+                {
+                    if (Mission.Current.InputManager.IsGameKeyDown(3))
+                        vec = new Vec3(1f, 0f, 0f, -1f);
+                    else
+                    {
+                        if (Mission.Current.InputManager.IsGameKeyDown(1))
+                            vec = new Vec3(0f, -1f, 0f, -1f);
+                        else
+                        {
+                            if (Mission.Current.InputManager.IsGameKeyDown(0))
+                                vec = new Vec3(0f, 1f, 0f, -1f);
+                        }
+                    }
+                }
+                blow.SwingDirection = agent.Frame.rotation.TransformToParent(vec);
+                blow.SwingDirection.Normalize();
+                blow.Direction = blow.SwingDirection;
+                blow.DamageCalculated = true;
+                agent.RegisterBlow(blow);
+            }
+        }
```

Comment:
This method looks like it's unused. If it is, I think we should go ahead and remove it.

Reply:
agree